python/tchannel/handler.py

```diff
@@ -76,14 +77,33 @@ def handle(self, context, connection):
         handler_name = "handle_" + self._HANDLER_NAMES[message.message_type]
         return getattr(self, handler_name)(message_id, message, connection)
 
+    def handle_pre_call(self, message_id, message, connection):
+        """Handle incoming request message including CallRequestMessage and
+        CallRequestContinueMessage
+
+        This method will build the User friendly request object based on the
+        incoming messages.
+
+        It passes all the messages into the message_factory to build the init
+        request object. Only when it get a CallRequestMessage and a completed
+        arg_1=argstream[0], the message_factory will return a request object.
+        Then it will trigger the async call_handle call.
+
+        :param message_id: message id
+        :param message: CallRequestMessage or CallRequestContinueMessage
+        :param connection: tornado connection
+        """
+        req = connection._request_message_factory.build(message_id, message)
```

Comment:
Accessing a protected attribute here.

Maybe this class should have its own message factory? Or perhaps assembly should happen somewhere else?